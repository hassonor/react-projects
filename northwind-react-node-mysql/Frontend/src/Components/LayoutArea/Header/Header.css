.Header {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}
