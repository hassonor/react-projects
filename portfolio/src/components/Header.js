import React, { useState } from "react";
import { RiMenu3Fill, RiCloseFill } from "react-icons/ri";
import { Link } from "react-router-dom";
function Header() {
  const [showHeader, setShowHeader] = useState(false);
  const path = window.location.pathname
  return (
    <div className="header">
      {showHeader ? (
        <RiCloseFill 
         onClick={()=>{setShowHeader(!showHeader)}}
        className="menu-icon position-fixed top-0 end-0" />
      ) : (
        <RiMenu3Fill className="menu-icon position-fixed top-0 end-0" onClick={()=>{setShowHeader(!showHeader)}}/>
      )}

      <ul className={`${showHeader ? 'show-header' : 'hide-header'} n-box1`}>
          <li className={`${path==='/' && 'active'}`}><Link to='/'>Home</Link></li>
          <li className={`${path==='/projects' && 'active'}`}><Link to='projects'>Projects</Link></li>
          <li className={`${path==='/contact' && 'active'}`}><Link to='/contact'>Contact</Link></li>
      </ul>
    </div>
  );
}

export default Header;
