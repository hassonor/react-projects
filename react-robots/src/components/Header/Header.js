import React from "react";
import CounterButton from "../CounterButton/CounterButton";

//Functional Component
const Header = React.memo((props) => {
  return (
    <>
      <h1 className="f1">Robots React</h1>
      <CounterButton />
    </>
  );
});

export default Header;
