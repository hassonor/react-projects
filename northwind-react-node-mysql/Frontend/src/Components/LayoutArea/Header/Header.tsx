import Logo from "../Logo/Logo";
import "./Header.css";

function Header(): JSX.Element {
    return (
        <div className="Header">
            <Logo />
            <h1>Northwind Traders</h1>
        </div>
    );
}

export default Header;