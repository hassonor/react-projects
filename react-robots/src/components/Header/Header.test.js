import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

it("expect to render Header Component", () => {
  expect(shallow(<Header />)).toMatchSnapshot();
});
