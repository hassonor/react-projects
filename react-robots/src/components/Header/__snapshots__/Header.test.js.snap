// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header Component 1`] = `
<Fragment>
  <h1
    className="f1"
  >
    Robots React
  </h1>
  <CounterButton />
</Fragment>
`;
